File: weather_script.js
// https://api.openweathermap.org/data/2.5/weather?q=London&appid=7
// 1. Crear una funcion que se encargue de conectarse a la api y obtener la informacion del clima

const fetchWeather = async (city) => {
    const apiKey = '08a7405286894c6d5fcc338cdb245207'; // Reemplaza con tu API key de OpenWeatherMap
    const apiUrl = `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}&units=metric`;

    try {
        const response = await fetch(apiUrl);
        if (!response.ok) {
            throw new Error(`Error: ${response.status} - ${response.statusText}`);
        }
        const weatherData = await response.json();
        console.log(`El clima en ${city} es:`, weatherData);
        return weatherData;
    } catch (error) {
        console.error('Hubo un problema al obtener el clima:', error.message);
    }
};

// Ejemplo de uso
fetchWeather('London');

File: weather_script.py
import requests

def fetch_weather(city_name, api_key):
    base_url = "http://api.openweathermap.org/data/2.5/weather"
    params = {
        "q": city_name,
        "appid": api_key,
        "units": "metric"  # Use "imperial" for Fahrenheit
    }
    try:
        response = requests.get(base_url, params=params)
        response.raise_for_status()  # Raise an HTTPError for bad responses
        data = response.json()
        return {
            "city": data["name"],
            "temperature": data["main"]["temp"],
            "description": data["weather"][0]["description"],
            "humidity": data["main"]["humidity"]
        }
    except requests.exceptions.RequestException as e:
        print(f"Error fetching weather data: {e}")
        return None

if __name__ == "__main__":
    API_KEY = "08a7405286894c6d5fcc338cdb245207"  # Replace with your OpenWeatherMap API key
    city = input("Enter the city name: ")
    weather_data = fetch_weather(city, API_KEY)
    if weather_data:
        print(f"City: {weather_data['city']}")
        print(f"Temperature: {weather_data['temperature']}°C")
        print(f"Weather: {weather_data['description']}")
        print(f"Humidity: {weather_data['humidity']}%")